### PR TITLE
Add troubleshooting for db access denied

### PR DESCRIPTION
**Pull request by GavinFrazar**
This is a docs only PR.

This PR adds troubleshooting info for "db access denied" errors. When a user gets this error, they will be directed to checking this page for troubleshooting.

### CODE REVIEW

**Review by zmb3 — APPROVED** [review]

File: docs/pages/database-access/troubleshooting.mdx [hunk 1]
```diff
@@ -39,3 +39,27 @@ details and examples:
...
+
+### Access to db denied
+
+Attempts to connect to the database fail with an error message
+similar to: **"access to db denied"**.
+
+**Solution:** Configure the database instance and Teleport role-based access control (RBAC) to allow the user 
+access.
+
+Check that the database user and database name (aka schema) exist in the database instance.
```
Comment (ptgott) [inline docs/pages/database-access/troubleshooting.mdx:51]:
```suggestion
Check that the database user and database name (aka schema) you are trying to access via `tsh` exist in the database instance.
```
To indicate where the database user and name come from (assuming this is correct)

File: docs/pages/database-access/troubleshooting.mdx [hunk 2]
```diff
@@ -39,3 +39,27 @@ details and examples:
...
+When referring to "database name" or `db_names`, this is not the same as the `name` field in `db_service`;
```
Comment (ptgott) [inline docs/pages/database-access/troubleshooting.mdx:52]:
```suggestion
When referring to "database name" or `db_names`, this is not the same as the `name` field in the `db_service` section of your Database Service configuration;
```

File: docs/pages/database-access/troubleshooting.mdx [hunk 3]
```diff
@@ -39,3 +39,27 @@ details and examples:
...
+this is referring to database names or schemas within a particular database instance.
+When connecting to a PostgreSQL or MongoDB instance, be sure to specify `--db-name`:
+```sh
+$ tsh db connect --db-user=exampleuser --db-name=exampledb
+```
+
+Once your database instance is configured, Teleport RBAC must be configured to allow access.
+Assign the Teleport user a role which **allows** `db_users`, `db_names`, and `db_labels` matching the 
+database user, database name, and Teleport static or dynamic labels for the database. Additionally, check that 
+the user does not have role(s) which **deny** corresponding `db_users`, `db_names`, or `db_labels`.
```
Comment (ptgott) [inline docs/pages/database-access/troubleshooting.mdx:62]:
```suggestion
the user does not have role(s) which **deny** the corresponding `db_users`, `db_names`, or `db_labels`.
```

File: docs/pages/database-access/troubleshooting.mdx [hunk 4]
```diff
@@ -39,3 +39,27 @@ details and examples:
...
+For example:
```
Comment (ptgott) [inline docs/pages/database-access/troubleshooting.mdx:63]:
```suggestion

Here is an example of creating a local Teleport user that can access any database user and database name:
```
Suggested making the connection to the preceding paragraph more explicit. I also think we should start this sentence in a new paragraph so it's more obvious that an example is coming.

File: docs/pages/database-access/troubleshooting.mdx [hunk 5]
```diff
@@ -39,3 +39,27 @@ details and examples:
...
+$ tsh db connect --db-user=exampleuser --db-name=exampledb
+```
+
+Once your database instance is configured, Teleport RBAC must be configured to allow access.
+Assign the Teleport user a role which **allows** `db_users`, `db_names`, and `db_labels` matching the 
+database user, database name, and Teleport static or dynamic labels for the database. Additionally, check that 
+the user does not have role(s) which **deny** corresponding `db_users`, `db_names`, or `db_labels`.
+For example: 
+
+(!docs/pages/includes/database-access/create-user.mdx!)
```
Comment (ptgott) [inline docs/pages/database-access/troubleshooting.mdx:65]: I'm not sure this partial really fits here. We don't need the flag descriptions, and the "Create a local Teleport user..." paragraph doesn't flow very smoothly from the preceding paragraph. What would you think of copying/pasting the partial and using only the relevant parts?

Comment (GavinFrazar) [reply docs/pages/database-access/troubleshooting.mdx]: I think you make good points. And actually, I think if a user gets directed to this page by an error then their Teleport user already exists - they don't need to create a new one.

I've updated the doc to provide a simple example in-line with a link to RBAC docs at the end

**Comment by GavinFrazar** [comment]
@ptgott Please take another look when you get the chance

File: docs/pages/database-access/troubleshooting.mdx [hunk 6]
```diff
@@ -39,3 +39,102 @@ details and examples:
...
+
+**Solution:** Configure the database instance and Teleport role-based access control (RBAC) to allow the user 
+access.
+
+Check that the database user and database name (aka schema) you are trying to access via `tsh` exist in the database instance.
+When referring to "database name" or `db_names`, this is not the same as the `name` field in the `db_service` section of your Database Service configuration;
+this is referring to database names or schemas within a particular database instance.
+
+Once your database instance is configured, Teleport RBAC must be configured to allow access.
+Assign the Teleport user a role which **allows** `db_users`, `db_names`, and `db_labels` matching the
```
Comment (ptgott) [inline docs/pages/database-access/troubleshooting.mdx:56]:
```suggestion
Assign the Teleport user a role that **allows** `db_users`, `db_names`, and `db_labels` matching the 
```

File: docs/pages/database-access/troubleshooting.mdx [hunk 7]
```diff
@@ -39,3 +39,102 @@ details and examples:
...
+access.
+
+Check that the database user and database name (aka schema) you are trying to access via `tsh` exist in the database instance.
+When referring to "database name" or `db_names`, this is not the same as the `name` field in the `db_service` section of your Database Service configuration;
+this is referring to database names or schemas within a particular database instance.
+
+Once your database instance is configured, Teleport RBAC must be configured to allow access.
+Assign the Teleport user a role which **allows** `db_users`, `db_names`, and `db_labels` matching the 
+database user, database name, and Teleport static or dynamic labels for the database. Additionally, check that 
+the user does not have role(s) which **deny** the corresponding `db_users`, `db_names`, or `db_labels`.
```
Comment (ptgott) [inline docs/pages/database-access/troubleshooting.mdx:58]:
```suggestion
the user does not have role(s) that **deny** the corresponding `db_users`, `db_names`, or `db_labels`.
```

File: docs/pages/database-access/troubleshooting.mdx [hunk 8]
```diff
@@ -39,3 +39,102 @@ details and examples:
...
+
+<Admonition type="warning">
+  Database names are only enforced for PostgreSQL and MongoDB databases.
+  When connecting to a PostgreSQL or MongoDB instance, be sure to specify `--db-name`:
+  ```sh
+  $ tsh db connect --db-user=exampleuser --db-name=exampledb
+  ```
+</Admonition>
+
+For example, the following Teleport user "alice@example.com"
```
Comment (ptgott) [inline docs/pages/database-access/troubleshooting.mdx:68]:
```suggestion
For example, the following `user` and `role` resources allow the Teleport user `alice@example.com` to
```
- Using code style so this doesn't show up as a link in the rendered paragraph
- Link this paragraph to the resources below more explicitly

File: docs/pages/database-access/troubleshooting.mdx [hunk 9]
```diff
@@ -39,3 +39,102 @@ details and examples:
...
+is permitted to access any database name or database user
```
Comment (ptgott) [inline docs/pages/database-access/troubleshooting.mdx:69]:
```suggestion
access any database name or database user
```
Needed if we accept my last suggestion

File: docs/pages/database-access/troubleshooting.mdx [hunk 10]
```diff
@@ -39,3 +39,102 @@ details and examples:
...
+within a production database except for the "admin" database user or
```
Comment (ptgott) [inline docs/pages/database-access/troubleshooting.mdx:70]:
```suggestion
within a production database except for the `admin` database user or
```

File: docs/pages/database-access/troubleshooting.mdx [hunk 11]
```diff
@@ -39,3 +39,102 @@ details and examples:
...
+the "postgres" database name:
```
Comment (ptgott) [inline docs/pages/database-access/troubleshooting.mdx:71]:
```suggestion
the `postgres` database name:
```

File: docs/pages/database-access/troubleshooting.mdx [hunk 12]
```diff
@@ -39,3 +39,102 @@ details and examples:
...
+```yaml
+kind: user
+metadata:
+  name: alice@example.com
+spec:
+  roles: ["db-developer"]
+  traits:
+    db_users: ["*"]
+    db_names: ["*"]
+```
```
Comment (ptgott) [inline docs/pages/database-access/troubleshooting.mdx:82]: Could we combine these two resources into the same YAML document using `---`?

**Review by ptgott — APPROVED** [review]
Approved with formatting, grammar, and style suggestions

File: docs/pages/database-access/troubleshooting.mdx (hunk 6 above)
Comment (GavinFrazar) [reply docs/pages/database-access/troubleshooting.mdx:56]: that/which strikes again!

File: docs/pages/database-access/troubleshooting.mdx (hunk 12 above)
Comment (GavinFrazar) [reply docs/pages/database-access/troubleshooting.mdx:82]: Possibly. I'll try it with `tctl create` and see if it complains about the role not existing yet - might need to change the order of the yaml documents.

Comment (GavinFrazar) [reply docs/pages/database-access/troubleshooting.mdx:82]: Ok we can combine them, but we do have to switch the document order.

**Pull request merged**